### ISSUE TITLE
Set "allow only sensible choices" to off by default & rename?

### ISSUE THREAD

**Issue by lodici** [magarena/magarena#343]
Originally raised by @PalladiaMors in #341.

> About "allow only sensible choices": it may be better to have that option off by default. I remember the first time I came across it I wanted to use Swords to Plowshares on one of my creatures and wasn't able to due to AI targeting restrictions. I had no idea that happened due to an option and thought it was a bug. The option's name is very counter-intuitive: it's very sensible to use swords to plowshares to gain life if you're about to die from damage. The human players know what they're doing when they target their own creatures with removal or damage or any other uncommon target choice, the restriction is really only required for the AI. The only advantage I see in having that option enabled is that it can speed up gameplay - but now Magarena has turn skipping, so that doesn't look as necessary anymore.

> Tl; dr: in my opinion it's best to have allow only sensible choices off by default, and it might be a good idea to find a better name for the option because uncommon uses of cards can definitely be sensible (I suggest "Enable casting restrictions for human players")?

**Comment by lodici** [comment]
I pretty sure that have been a number of cases where the response to an issue has been "is the 'allow only sensible choices' option enabled?". That does seem to suggest that it may need to be made clearer when it is on/off during a game (maybe an icon? somewhere). As to its name - I remember having quite a discussion about this when we renamed it to what it is now, although I quite like your suggestion.

**Comment by ShawnieBoy** [comment]
I'm certainly in favour of having off as default, it's the first thing I turn off!

@PalladiaMors raises a good point as well with the turn skipping making it less 'useful' - there could be a move to remove it completely. The only use I could think for it would be testing the AI targeting restrictions when scripting new cards (which I don't do, but could be something I could consider for niche cases)

**Comment by melvinzhang** [comment]
Agree that with the more noticeable skip till EOT option, having the limits off as default is better and less confusing. I'm thinking perhaps it can be repurpose as kind of difficulty setting "Limit options for human player to those available to the AI". It can be considered an option that makes the game more fair and challenging by limiting the both players to the same set of options.

**Comment by PalladiaMors** [comment]
That's a perspective that I hadn't really considered. It is pretty unbalancing to allow one player to do things that the other is completely prevented from doing. Given this, it actually might be better to leave it on by default? It's kind of treating it as a house rule from Magarena, to keep things fair.

**Comment by lodici** [comment]
I think first and foremost this setting when on by default leads to a usability issue whereby a player is unable to perform an action that they would reasonably expect to work. Since there is no indication on the duel screen that the "sensible" setting is in effect this leads to confusion.

Then there is the setting itself which seems to have two roles as far as the human player is concerned. 

The first is to "help" a player only make sensible decisions but that seems fraught with subjective interpetation as to what constitutes "sensible" and surely cannot take into account all the myriad permutations that the game board state can be in at any one time. My view is that this particular restriction be removed for the human player, if a mistake is made then so what, it's all part and parcel of the game. We learn from our mistakes and there is always the undo option.

The second use of this setting is to automatically pass priority if there is no legal play available. This seems like a very useful option and can speed up game play but I think this should either be on by default or if optional then a specific setting should be added, maybe something like "auto pass if no play available" (although I cannot see why you would ever turn this off). 

Finally, a comment on the idea that there should be some sort of "equality" and "fairness" for both players, human and AI. I think what we all desire is an AI opponent that plays a strong and challenging game. When the AI makes an obviously stupid or nonsensical move it spoils the immersion. If this can be alleviated by letting the AI cheat now and again then why not. I think you will find most games that are considered to have strong AI are a compromise between pure AI "learning" and plain old hacks.

**Comment by melvinzhang** [comment]
I think primarily the setting is there to speed up gameplay by reducing the set of possible plays. This together with "Automatically pass priority" (automatically pass if pass is the only option) makes for faster gameplay

> I think first and foremost this setting when on by default leads to a usability issue whereby a player is unable to perform an action that they would reasonably expect to work. 

Agree, it should be off by default now that players can use "skip till EOT" to advance the game. The "speed up gameplay" reason is no longer valid.

> The second use of this setting is to automatically pass priority if there is no legal play available. 

Are you talking about the "Automatically pass priority" option? It is on by default. There is a certain synergy between the two, "allow only sensible choices" reduces the options available, "Automatically pass priority" passes automatically without player interaction if pass is the only option. Having the former enabled, means the latter gets happens more often.

> When the AI makes an obviously stupid or nonsensical move it spoils the immersion. If this can be alleviated by letting the AI cheat now and again then why not.

The way we are reducing the nonsensical moves is not by allowing the AI to cheat, but rather by reducing the AI's choices to what we think are sensible moves.

However, this introduces a disparity, as the AI is limited to making the "sensible" moves, while the human player is able to make any move. It also means the AI is unable to anticipate such unexpected moves when it is planning ahead in the game.

Hence my suggestion for players who wish for a greater challenge to give themselves a handicap by enabling a "Limit options for human player to those available to the AI" option.

**Comment by ShawnieBoy** [comment]
:+1: 
>Hence my suggestion for players who wish for a greater challenge to give themselves a handicap by enabling a "Limit options for human player to those available to the AI" option.

This looks like the best possible solution for the 'sensible' choices option. Off as default and renamed to the above.

**Comment by lodici** [comment]
> Hence my suggestion for players who wish for a greater challenge to give themselves a handicap by enabling a "Limit options for human player to those available to the AI" option.

I still think this is a very vague, hard to pin down setting. Can you state specifically how it will affect gameplay such that when, as an average player with no knowledge (or interest in) how AI works, I select this option I will be clearly aware when it is affecting gameplay? Instead of thinking, _Eh? why is it doing that or why is not letting me do this? Better raise a bug report..._

**Comment by ShawnieBoy** [comment]
Maybe adding/editing the description of the option to explain what it does. As it would be off by default, users would have to enable it, and so see the description.

**Comment by melvinzhang** [comment]
Agree with @ShawnieBoy the description would be a good place to put the detailed explanation. A possible description is 

"Positive effects, eg. pump, gain ability, can only be applied to your own creatures. 
 Negative effects, eg. destroy, exile, can only be applied to opponent's permanents"

I think it would be quite clear if after enabling a difficulty option and then something is different in a game to attribute it to the option you just turned on.
